src/mlpack/methods/ann/init_rules/glorot_init.hpp

```diff
@@ -0,0 +1,145 @@
+ * and back-propagated gradients variance as one moves up or down the network.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_ANN_INIT_RULES_GLOROT_INIT_HPP
+#define MLPACK_METHODS_ANN_INIT_RULES_GLOROT_INIT_HPP
+
+#include <mlpack/prereqs.hpp>
+#include "random_init.hpp"
+#include "gaussian_init.hpp"
+
+using namespace mlpack::math;
+
+namespace mlpack {
+namespace ann /** Artificial Neural Network. */ {
+
+/**
+ * This class is used to initialize the weight matrix with the Glorot Initialization
+ * method. The method is defined by
+ *
+ * @f{eqnarray*}{
+ * \f$\Var[w_i] &=& \frac{2}{n_i + n_{i+1}} \\\f$
+ * \f$w_i ~ \U{-\frac{\sqrt(6)}{\sqrt(n_i + n_{i+1})}, \frac{\sqrt(6)}{\sqrt(n_i + n_{i+1})}\f$}
+ * @f}
+ * Where n_{i+1} is the number of neurons in the outgoing layer, n_i represents the
+ * number of neurons in the ingoing layer
+ * Here Normal Distribution may also be used if needed
+ *
+ * For more information, see the following paper.
+ *
+ * @code
+ * @inproceedings {pmlr-v9-glorot10a,
+ *  title       = {Understanding the difficulty of training deep feedforward neural networks},
+ *  author      = {Xavier Glorot and Yoshua Bengio},
+ *  booktitle   = {Proceedings of the Thirteenth International Conference on Artificial
+ *                 Intelligence and Statistics},
+ *  year        = {2010}
+ * }
+ * @endcode
+ *
+ */
+template<bool Uniform = true>
+class GlorotInitializationType
+{
+ public:
+  /**
+   * Initialize
+   */
+  GlorotInitializationType()
+  {
+    // Nothing to do here.
+  }
+
+  /**
+   * Initialize the elements weight matrix.
+   *
+   * @param W Weight matrix to initialize.
+   * @param rows Number of rows.
+   * @param cols Number of columns.
+   */
+  template<typename eT>
+  void Initialize(arma::Mat<eT>& W,
+                  const size_t rows,
+                  const size_t cols);
+
+  /**
+   * Initialize the elements of the specified weight 3rd order tensor with glorot initialization method
+   *
+   * @param W Weight matrix to initialize.
+   * @param rows Number of rows.
+   * @param cols Number of columns.
+   * @param slice Numbers of slices.
+   */
+  template<typename eT>
+  void Initialize(arma::Cube<eT>& W,
+                  const size_t rows,
+                  const size_t cols,
+                  const size_t slices);
+}; // class GlorotInitializationType
+
+template <>
+template<typename eT>
+inline void GlorotInitializationType<false>::Initialize(arma::Mat<eT>& W,
+                                                       const size_t rows,
+                                                       const size_t cols)
+{
+  if (W.is_empty())
+  W = arma::mat(rows, cols);
+
+  double_t var = (double)(2)/(rows + cols);
```

Comment:
Can you avoid the use of `double_t` and use `double` instead throughout the code please?  I might suggest rewriting the line like this:

```
double var = 2.0 / double(rows + cols);
```